### ISSUE TITLE
Problem with proc macro contract! (on Windows)

### ISSUE THREAD

**Issue by PraetorP** [cowprotocol/ethcontract-rs#729]
Hi !
I'm recently working with Eth.. I downloaded your sources and wanted to see how the examples work, but I ran into the fact that this macro cannot find files with ABI (The filename, directory name, or volume label syntax is incorrect. (os error 123)) . I tried replacing "/" with "\\" but that didn't help. Can you tell me what I'm doing wrong? Thanks ! )

From under WSL the code is compiled.
![image](https://user-images.githubusercontent.com/45266194/156751273-0d7a5788-455c-4fe1-addf-1e5ab3290ae9.png)
![image](https://user-images.githubusercontent.com/45266194/156751321-9fa1af7b-891a-4fe6-8918-68eac59e4e2d.png)

**Comment by nlordell** [comment]
Hmm, it could be an issue with Windows-specific path things. Unfortunately, I don't have a Windows machine to test this out and fix it :'(.

One thing you might be able to do is use the `ethcontract-generate` crate instead of the proc macro?

**Comment by PraetorP** [comment]
Thank you very much for your answer! As I understand it, the problem only exists with local paths.
Can you please tell me how to get access to the methods of the smart contract, with such ABI initialization?
![image](https://user-images.githubusercontent.com/45266194/156766118-9a556e2e-281d-4961-9a4a-4ff8e7179ab5.png)

**Comment by nlordell** [comment]
You should be able to use an "etherscan source":

```rust
ethcontract::contract!(
    "etherscan:0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D",
    contract = UniswapV2Router02,
    mod = router,
);
```

**Comment by PraetorP** [comment]
Thanks ! It works ! The only thing, not quite stable. Sometimes the "etherscan..." line may turn red.
Unfortunately, I'm not familiar with macros in Rust yet. To be honest, I would like to understand what exactly leads to such behavior under Win ;)

**Comment by nlordell** [comment]
> Sometimes the "etherscan..." line may turn red

This could be a LSP (`rls` or `rust-analyzer`) thing - I know they sometimes have issues with procedural macros (in one of my projects `derivative::Derivative` ALWAYS shows errors from `rust-analyzer` but doesn't have any issues building).

If this is bothering you - you can essentially do the exact same thing your proc-macro is doing with a `build.rs` script and the `ethcontract-generate` crate (including downloading the ABI from Etherscan - although I would recommend vendoring the ABI). See the `generate` example on how that works.

**Comment by rayeaster** [comment]
same issue on windows path. have to use other source like npm or etherscan. guess it might help others to resolve this issue if possible
@nlordell @Jaxkr

**Comment by nlordell** [comment]
@rayeaster - would it be possible for you to try out with `main` branch instead of the last released version?

Specifically, there has been no new release with #781 included.

If this fixes things for you, I will make a 0.17.1 release.

**Comment by rayeaster** [comment]
> @rayeaster - would it be possible for you to try out with `main` branch instead of the last released version?
> 
> Specifically, there has been no new release with #781 included.
> 
> If this fixes things for you, I will make a 0.17.1 release.

could you please shed a light on how to compile an executable for Windows from the `main` branch? I am using `ethcontract` in `cowswap-solver` https://github.com/gnosis/cow-dex-solver

**Comment by nlordell** [comment]
You should be able to add a `[patch]` section in your `Cargo.toml` file to override the `ethcontract` dependency version:

https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section

Something like:
```toml
[patch.crates-io]
ethcontract = { git = 'https://github.com/gnosis/ethcontract-rs' }
```
